lib/submit-analytics.js

```
@@ -33,50 +104,184 @@ function sha256(data) {
+                    // on v6.x, we submit both MixPanel and S3 analytics
+                    // MixPanel analytics make no distinction between production and staging modes
+                    dispatchAnalytics(new mixpanelDetails(), payload);
+                }
+
+                dispatchAnalytics(requestFactory, payload)
+                .catch((e) => {
+                    reject(e);
+                    return;
+                })
+                .then(() => {
+                    resolve('');
+                    return;
                 });
+            });
         }, 100);
-    }
 
-    if (require.main === module) {
-        module.exports('Install');
-    }
+    }); // end Promise
+}
+
+/**
+ * Shorthand for submitting analytics to production environment
+ * @param  {string} eventName Custom tag that signifies what the app was doing when the analytics were submitted
+ * @returns Promise
+ */
+async function submitProductionAnalytics(eventName) {
+    const context = require('../package.json')
+    return submitAnalytics(webhookProdDetails, context, eventName);
+}
+
+/**
+ * Shorthand for submitting analytics to staging environment
+ * @param  {string} eventName Custom tag that signifies what the app was doing when the analytics were submitted
+ * @returns Promise
+ */
+async function submitStageAnalytics(eventName) {
+    const context = require('../package.json');  // Realm context
+    return submitAnalytics(webhookStageDetails, context, eventName);
+}
+
+module.exports.fetchPlatformData = fetchPlatformData;
+module.exports.webhookProdDetails = webhookProdDetails;
+module.exports.webhookStageDetails = webhookStageDetails;
+module.exports.submitAnalytics = submitAnalytics;
+module.exports.submitProductionAnalytics = submitProductionAnalytics;
+module.exports.submitStageAnalytics = submitStageAnalytics;
+module.exports.isAnalyticsDisabled = isAnalyticsDisabled;
+
+if (require.main === module) {
+    module.submitProductionAnalytics('Install');
```

Either warn the user about this or simply "swallow" this error. We need to catch a rejected promise never the less, or it might end up as an unhandled promise rejection.

```suggestion
    module.submitProductionAnalytics('Install').catch(err => {
        console.warn(`Failed to submit analytics: ${err.message}`);
    });
```

---

Done